plugin/storage/es/factory.go

```diff
@@ -133,3 +141,42 @@ func loadTagsFromFile(filePath string) ([]string, error) {
 	}
 	return tags, nil
 }
+
+// CreateArchiveSpanReader implements storage.ArchiveFactory
+func (f *Factory) CreateArchiveSpanReader() (spanstore.Reader, error) {
```

Comment:
can we DRY this to be written only once based on the passed cfg?

Reply:
yes this is just a WIP